controllers/statusreport/statusreport_adapter.go

```diff
@@ -86,17 +90,42 @@ func (a *Adapter) EnsureSnapshotTestStatusReportedToGitProvider() (controller.Op
 			return controller.RequeueWithError(err)
 		}
 	}
+	integrationTestScenarios, err := a.loader.GetRequiredIntegrationTestScenariosForApplication(a.client, a.context, a.application)
```

Comment:
If here we call `GetRequiredIntegrationTestScenariosForApplication()` to remove finalizer from the PLR of required ITS, should we also handle the PLR triggered by other ITS in all ITS?
In snapshot_adapter.go [EnsureStaticIntegrationPipelineRunsExist()](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter.go#L192-L204), we report status for all ITS to snapshot annotation, so should we change to call `GetAllIntegrationTestScenariosForApplication()` here?

Reply:
We don't even need that function, we can just iterate over test statuses

Reply:
most recent changes uses @MartinBasti suggestion to just use the existing statuses